### PR TITLE
New formula: Prometheus

### PR DESCRIPTION
**Pull request by anthonygreen**

### CODE REVIEW

File: Library/Formula/prometheus.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+class Prometheus < Formula
+  desc "Service monitoring system and time series database"
+  homepage "http://prometheus.io"
```
Comment (DomT4) [inline Library/Formula/prometheus.rb]: This can be `https://prometheus.io/`

File: Library/Formula/prometheus.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  url "https://github.com/prometheus/prometheus.git",
+    :tag => "0.17.0",
+    :revision => "e11fab35d76d19c5c49b7d85e28275f894d3ada4"
+  sha256 "363400cd2e1167a1a62d99fc344b477c97c289876d70d39318d82855b6eaf2c2"
```
Comment (DomT4) [inline Library/Formula/prometheus.rb]: The `:revision` above acts as the checksum so you don't need this line.

File: Library/Formula/prometheus.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    ENV["GO15VENDOREXPERIMENT"] = "1"
+    mkdir_p buildpath/"src/github.com/prometheus"
+    ln_sf buildpath, buildpath/"src/github.com/prometheus/prometheus"
+
+    system "make", "build"
+    bin.install "promtool"
```
Comment (DomT4) [inline Library/Formula/prometheus.rb]: Use a `%w[]` array to avoid using two lines here.

File: Library/Formula/prometheus.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    bin.install "prometheus"
+    libexec.install "consoles"
```
Comment (DomT4) [inline Library/Formula/prometheus.rb]: And here.

File: Library/Formula/prometheus.rb [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    libexec.install "console_libraries"
+  end
+
+  test do
+    system "#{bin}/prometheus", "--version"
```
Comment (DomT4) [inline Library/Formula/prometheus.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

**Comment by anthonygreen** [comment]
I wasn't able to figure out how to improve the test for the prometheus binary even after looking at cmake.rb and tinyxml2.rb. If someone could submit a pull request I'll merge it.

File: Library/Formula/prometheus.rb [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    mkdir_p buildpath/"src/github.com/prometheus"
+    ln_sf buildpath, buildpath/"src/github.com/prometheus/prometheus"
+
+    system "make", "build"
+    bin.install %w[promtool prometheus]
+    libexec.install %w[consoles console_libraries]
+  end
+
+  test do
+    system "#{bin}/prometheus", "--version"
```
Comment (mikemcquaid) [inline Library/Formula/prometheus.rb]: You can remove this line given the better test below.

**Comment by mikemcquaid** [comment]
@anthonygreen Other than the comment: the test (and PR) looks good to me :+1:

**Comment by anthonygreen** [comment]
@mikemcquaid done

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format for simple version updates is `prometheus 0.17.0 (new formula)`. Please leave complete the PR template (which links to the contributing guide detailing this) in future, thanks!

**Pull request closed without merging**